tools/proofers/hiero/index.php

```diff
@@ -2,15 +2,201 @@
 $relPath="./../../../pinc/";
 include_once($relPath.'base.inc');
 include_once($relPath.'slim_header.inc');
+include_once($relPath.'misc.inc'); // html_safe()
+include_once("$wikihiero_dir/wh_language.php");
+include_once("$wikihiero_dir/wikihiero.php");
 
 require_login();
 
-slim_header_frameset(_("Hieroglyphs"));
+$header_args = array(
+  'js_files' => array(
+      "$code_url/tools/proofers/hiero/hiero.js",
+  ),
+);
+
+slim_header(_("Hieroglyphs"), $header_args);
+
+$tables=array(
+    "Phoneme",
+    "A",
+    "B",
+    "C",
+    "D",
+    "E",
+    "F",
+    "G",
+    "H",
+    "I",
+    "K",
+    "L",
+    "M",
+    "N",
+    "O",
+    "P",
+    "Q",
+    "R",
+    "S",
+    "T",
+    "U",
+    "V",
+    "W",
+    "X",
+    "Y",
+    "Z",
+    "Aa",
+    "All"
+);
+
+$syntax=array("-",":","*","!");
+if(array_key_exists("table", $_POST))
+    $table = $_POST["table"];
+else
+    $table = "";
+?>
+<form name="hieroform" method="POST">
+<table>
+<tr><td rowspan="2">
+<textarea name="hierobox" rows="4" cols="30">
+<?php $hierobox=@$_POST['hierobox']; echo html_safe($hierobox); ?>
+</textarea>
+</td><td colspan="2">
+<select onChange="hieroform.submit()" name="table">
+<?php
+echo "<option>".html_safe(WH_Text("Tables"))."</option>";
```

Comment:
Disclaimer: I know this is simply a lift from what was there before.

Rather than have this interlaced php -> html -> php -> html -> php, consider building the select options into a string first, then just echo'ing the value. Even better would be to use a heredoc for the HTML so the entire files is php:

```php
echo <<<HTML
<form name="hieroform" method="POST">
<table>
<tr><td rowspan="2">
<textarea name="hierobox" rows="4" cols="30">
$escaped_hierobox
</textarea>
</td><td colspan="2">
<select onChange="hieroform.submit()" name="table">
$table_options
</select>
...
HTML;
```

Ideally, shove the select-generating logic (and the `$tables` and `$syntax` arrays) into two functions after the HTML blob, but I'm OK leaving that logic at the top-level of the file too.